eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -59,47 +60,47 @@
     public Resource getCoreInfo(@PathVariable("variantCoreString") String variantCoreString,
                                 @RequestParam(name = "species") String species,
                                 @RequestParam(name = "assembly") String assembly,
-                                HttpServletResponse response) throws IllegalArgumentException{
+                                HttpServletResponse response) throws IllegalArgumentException {
         initializeQuery();
         try {
             checkParameters(variantCoreString, null, null, species, assembly);
         } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return new Resource<>(setErrorQueryResponse(e.getMessage()));
         }
 
         MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
 
-        List<VariantWithSamplesAndAnnotation> variantEntities;
-        Long numTotalResults;
+        Optional<VariantWithSamplesAndAnnotation> variantEntity;
 
         try {
-            variantEntities = getVariantEntitiesByParams(variantCoreString, null, null);
-            numTotalResults = (long) variantEntities.size();
-            if (variantEntities.size() == 0) {
-                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-                return new Resource<>(setQueryResponse(variantEntities));
-            }
+            variantEntity = getVariantByCoordinatesAndAnnotationVersion(variantCoreString, null, null);
         } catch (AnnotationMetadataNotFoundException ex) {
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
             return new Resource<>(setQueryResponse(ex.getMessage()));
+        } catch (IllegalArgumentException ex) {
```

Comment:
If multiple exception types are treated the same way, Java allows to capture them all as described [here](http://tutorials.jenkov.com/java-exception-handling/catching-multiple-exceptions.html).

Reply:
Fixed :+1: